tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -237,4 +222,40 @@ object NativeConfig {
     }
   }
 
+  def predefinedLinktimeProperties(config: NativeConfig): LinktimeProperites = {
+    val linktimeInfo = "scala.scalanative.meta.linktimeinfo"
+    Map(
+      s"$linktimeInfo.isWindows" -> Platform.isWindows,
+      s"$linktimeInfo.isWeakReferenceSupported" -> {
+        config.gc == GC.Immix ||
+        config.gc == GC.Commix
+      }
+    )
+  }
+
+  def checkLinktimeProperties(properties: LinktimeProperites): Unit = {
+    def isNumberOrString(value: Any) = {
+      def hasSupportedType = value match {
+        case _: Boolean | _: Byte | _: Char | _: Short | _: Int | _: Long |
+            _: Float | _: Double | _: String =>
+          true
+        case _ => false
+      }
+
+      value != null && hasSupportedType
+    }
+
+    val invalid = properties.collect {
+      case (key, value) if !isNumberOrString(value) => key
+    }
+    if (invalid.nonEmpty) {
+      System.err.println(
+        s"Invalid link-time properties: \n ${invalid.mkString(" - ", "\n", "")}"
+      )
```

Comment:
This should be part of the exception message, instead of being printed to the console.